src/raqm.h

```diff
@@ -76,25 +74,25 @@ typedef enum
 
 /**
  * raqm_glyph_t:
+ * @hbfont: the harfbuzz font of the glyph.
  * @index: the index of the glyph in the font file.
  * @x_advance: the glyph advance width in horizontal text.
  * @y_advance: the glyph advance width in vertical text.
  * @x_offset: the horizontal movement of the glyph from the current point.
  * @y_offset: the vertical movement of the glyph from the current point.
  * @cluster: the index of original character in input text.
- * @ftface: the @FT_Face of the glyph.
  *
  * The structure that holds information about output glyphs, returned from
  * raqm_get_glyphs().
  */
 typedef struct raqm_glyph_t {
+    hb_font_t* hbfont;
```

Comment:
structure size opt: putting the large pointer first in the structure.

Reply:
This changes the API. The bigger question is how we handle this without breaking both the API and ABI? I don’t have any good ideas.

Reply:
I don't think there's use case of using both api's at the same time, I would opt to use the ifdef like so:
```
 typedef struct
 {
#ifdef RAQM_FREETYPE
   FT_Face       ftface;
   int           ftloadflags;
#else
  hb_font_t    *hbfont;
#endif
```

What do you think?